draco/analysis/beamform.py

```diff
@@ -607,6 +609,10 @@ def _process_catalog(self, catalog):
         if "position" not in catalog:
             raise ValueError("Input is missing a position table.")
 
+        if not hasattr(self, "epoch"):
+            self.log.warning("Epoch not set. Was the requested data not available?")
+            self.data_available = False
```

Comment:
We need to return here.

Reply:
Should probably do the check in `process_data`, bc that is where "no data being available" is discovered.

And return non-zero if no data available, or something.

Reply:
We are going to stick with the current change because of this bug: https://github.com/radiocosmology/caput/issues/181#issuecomment-924398276

For `BeamFormCat`, `_process_data` in `setup()`, and so it will not be called, if there is no data available.

Reply:
I guess this one will probably need some work to merge with the current version.

Reply:
Sigh. Yeah.

Reply:
Same here.